generator-preview-utils/wrap-text-canvas.ts

```diff
@@ -15,67 +15,41 @@ export const wrapText = function (
   lineHeight: number
 ) {
   // First, start by splitting all of our text into words, but splitting it into an array split by spaces
-  let words = text.split(' ')
+  let words = text.split(/\s+/)
   let line = '' // This will store the text of the current line
-  let testLine = '' // This will store the text when we add a word, to test if it's too long
-  let lineArray = [] // This is an array of lines, which the function will return
+  let lineArray: [string, number, number][] = [] // This is an array of lines, which the function will return
+                                                 // [text, x, y] 
 
   // Lets iterate over each word
-  for (var n = 0; n < words.length; n++) {
+  for (const word of words) {
     // Create a test line, and measure it..
-    testLine += `${words[n]} `
-    let metrics = ctx.measureText(testLine)
+    let metrics = ctx.measureText(line + ' ' + word)
     let testWidth = metrics.width
-    // If the width of this test line is more than the max width
-    if (testWidth > maxWidth && n > 0) {
+    // If the width of this line is more than the max width
+    if (testWidth > maxWidth && ctx.measureText(word).width <= maxWidth) {
```

Comment:
the issue of words bigger than the maxWidth does not only occurs for the first words